planning/obstacle_stop_planner/src/planner_utils.cpp

```diff
@@ -449,6 +449,32 @@ bool withinPolygon(
   return find_within_points;
 }
 
+bool withinPolyhedron(
+  const std::vector<cv::Point2d> & cv_polygon, const double radius, const Point2d & prev_point,
+  const Point2d & next_point, PointCloud::Ptr candidate_points_ptr,
+  PointCloud::Ptr within_points_ptr, double z_min, double z_max)
+{
+  Polygon2d boost_polygon;
+  bool find_within_points = false;
+  for (const auto & point : cv_polygon) {
+    boost_polygon.outer().push_back(bg::make<Point2d>(point.x, point.y));
+  }
+  boost_polygon.outer().push_back(bg::make<Point2d>(cv_polygon.front().x, cv_polygon.front().y));
+
+  for (size_t j = 0; j < candidate_points_ptr->size(); ++j) {
+    Point2d point(candidate_points_ptr->at(j).x, candidate_points_ptr->at(j).y);
+    if (bg::distance(prev_point, point) < radius || bg::distance(next_point, point) < radius) {
+      if (bg::within(point, boost_polygon)) {
+        if (candidate_points_ptr->at(j).z < z_max && candidate_points_ptr->at(j).z > z_min) {
+          within_points_ptr->push_back(candidate_points_ptr->at(j));
```

Comment:
```suggestion
  for (const auto & p : *candidate_points_ptr) {
    Point2d point(p.x, p.y);
    if (bg::distance(prev_point, point) < radius || bg::distance(next_point, point) < radius) {
      if (bg::within(point, boost_polygon)) {
        if (p.z < z_max && p.z > z_min) {
          within_points_ptr->push_back(p);
```

Reply:
@maxime-clem We are planning to run the test in AWSIM to confirm that the original issue is solved and record some videos. But to do that we need a bit more time to create a new ITRI binary based on humble instead of galactic. I will be working on it next week.

Reply:
@maxime-clem 
Thank you for your review. I've applied your hints.